src/mscorlib/shared/System/Security/SecurityException.cs

```diff
@@ -45,12 +54,26 @@ public SecurityException(string message, Type type, string state)
         protected SecurityException(SerializationInfo info, StreamingContext context)
             : base(info, context)
         {
-            throw new PlatformNotSupportedException();
+            Demanded = (string)info.GetValueNoThrow(DemandedName, typeof(string));
+            GrantedSet = (string)info.GetValueNoThrow(GrantedSetName, typeof(string));
+            RefusedSet = (string)info.GetValueNoThrow(RefusedSetName, typeof(string));
+            DenySetInstance = (string)info.GetValueNoThrow(DeniedName, typeof(string));
+            PermitOnlySetInstance = (string)info.GetValueNoThrow(PermitOnlyName, typeof(string));
+            Url = (string)info.GetValueNoThrow(UrlName, typeof(string));
         }
 
         public override string ToString() => base.ToString();
 
-        public override void GetObjectData(SerializationInfo info, StreamingContext context) => base.GetObjectData(info, context);
+        public override void GetObjectData(SerializationInfo info, StreamingContext context)
+        {
+            base.GetObjectData(info, context); 
```

Comment:
If you have to push again for some reason, it would be nice to have a comment explaining that we are intentionally dropping fields X, Y Z

Reply:
ok